### ISSUE TITLE
fix remote repo teardown

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1370]
When quickly starting a lot of (30) borg (1.0-maint branch) against same repository via ssh (to localhost), I frequently see this, but the majority of tasks work:

```
Remote: ssh_exchange_identification: Connection closed by remote host
Exception ignored in: <bound method RemoteRepository.__del__ of <RemoteRepository ssh://tw@localhost/home/tw/w/borg/repo>>
Traceback (most recent call last):
  File "/home/tw/w/borg/borg/remote.py", line 184, in __del__
    assert False, "cleanup happened in Repository.__del__"
AssertionError: cleanup happened in Repository.__del__
Connection closed by remote host. Is borg working on the server?
terminating with error status, rc 2
```

**Comment by enkore** [comment]
Guess [here](https://github.com/borgbackup/borg/blob/1.0-maint/borg/remote.py#L168) are also explicit close calls needed, like in L177.

**Comment by ThomasWaldmann** [comment]
Yup, that's helping. Not that ugly any more.

The remaining question is of course why it is "closed by remote" in some cases before it even started.

**Comment by ThomasWaldmann** [comment]
Hmm, a general problem when first closing then (re-)raising is that the channel to the client is closed then (at least in cases when it is not closed due to other reasons already), so the exception logging won't get through to the client.

**Comment by ThomasWaldmann** [comment]
Fixed by #1408.
